adapters/sinatra/lib/opentelemetry/adapters/sinatra/adapter.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'sinatra'
+require 'opentelemetry'
+
+require_relative 'extensions/tracer_extension'
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      class Adapter
+        class << self
+          attr_reader :config,
+                      :propagator
+
+          def install(config = {})
+            return :already_installed if installed?
+
+            @config = config
+            @propagator = tracer_factory.http_text_format
+
+            new.install
+          end
+
+          def tracer
+            @tracer ||= OpenTelemetry.tracer_factory.tracer(
+              Sinatra.name,
+              Sinatra.version
+            )
+          end
+
+          attr_accessor :installed
+          alias_method :installed?, :installed
+
+          private
+
+          def tracer_factory
+            OpenTelemetry.tracer_factory
+          end
+        end
+
+        def install
+          return :already_installed if self.class.installed?
+          self.class.installed = true
+
+          register_tracer_extension
+
+          :installed
```

Comment:
I'm probably missing a bunch of earlier discussion. What is the intent behind the `:installed` and `:already_installed` return values? I don't see the return value used in the example.

Reply:
Discussed IRL: this is mostly documentation & makes testing more convenient.